### DISCUSSION TITLE
UAVSAR reference xml

### DISCUSSION THREAD

**Discussion by ZachHoppinen** [isce-framework/isce2#469] [Q&A] [answered]
Hello,

I am interested in processing pairs of UAVSAR images (either from an UAVSAR SLC stack or from PolSAR images) to form interferograms. I was hoping to find an appropriate XML file example in the examples/input_files directory but didn't see one in there. The only file was isceappUAVSAR_Stack.xml which seems to go with a polarimetric analysis of ground projected files from the tutorial on the ISCE2-docs (chapter 12). Is there another reference XML file for uavsar files somewhere?

Or possibly the InSAR app doesn't support UAVSAR? I just assumed it did because of ReadMe: "At this time the sensors that are supported are the following: ALOS, ALOS2, COSMO_SKYMED, ENVISAT, ERS, KOMPSAT5, RADARSAT1, RADARSAT2, RISAT1, Sentinel1, TERRASARX, UAVSAR and SAOCOM1A."

Thanks for your help!

**Comment by ZachHoppinen** [comment]
It seems like based on: https://github.com/isce-framework/isce2/blob/def109815df5f724c5dc653878314577e3505c48/components/isceobj/Sensor/UAVSAR_Stack.py that my reference and activity index may need an: "annotation file" component and a "segment index" component. The annotation file part makes sense but does anyone have any clues as to what segment index is?

**Comment by EJFielding** [comment, marked as answer]
I assume you are you are trying to process UAVSAR coregistered SLC stack data with ISCE2. You should use the isce2/contrib/stack/stripmapStack workflow that is designed to work with coregistered SLC images. 

The Jupyter notebook by @forrestfwilliams https://github.com/forrestfwilliams/UAVSAR_InSAR describes the workflow applied to UAVSAR SLC stack data.

**Reply by ZachHoppinen** [reply]
Hey Eric,

Thanks for the reply! This is very helpful.

This seems like a really helpful repo and I assume you are specifically talking about this [notebook](https://github.com/forrestfwilliams/UAVSAR_InSAR/blob/main/interferogram_processing.ipynb).

Thank you for your time!
Zach

**Reply by EJFielding** [reply]
That script should be in isce2/contrib/stack/stripmapStack

**Reply by ZachHoppinen** [reply]
Gotcha I see [it](https://github.com/isce-framework/isce2/blob/main/contrib/stack/stripmapStack/prepareUAVSAR_coregStack.py) now. I am on UAF (Fairbank)'s  OpenSarLab and the version of isce they have installed doesn't seem to have that script on it. I am gonna check with the folks up there and see why their version of isce looks different.

**Reply by forrestfwilliams** [reply]
Thanks for passing the notebook along @EJFielding. @ZachKeskinen, feel free to reach out if you have any questions about the notebook.

**Reply by ZachHoppinen** [reply]
Hey Forrest and Eric,

Thanks! That is huge to have a reference notebook to work from. I have gotten all the steps up to the interferogram working and am currently having the computer crunch through step 8. I really appreciate the helpful point in the right direction and the great example notebook.

**Reply by ZachHoppinen** [reply]
Got it to work! Thanks guys! One thing I was wondering, I wasn't able to use snaphu to unwrap and had to swap to icu due to memory overload. Is there a reason to use one over the other?

**Reply by EJFielding** [reply]
Yes, the SNAPHU program can use a huge amount of memory if you run it on a large image. It loads the whole interferogram and coherence map into memory and makes additional copies in memory for the unwrapping algorithm. I normally use 3 range and 12 azimuth looks to get roughly 5 meter pixels for the UAVSAR interferograms before unwrapping. That means it can run with about 5-10 GB of RAM. Alexander Handwerger at JPL was recently processing some UAVSAR with fewer looks. I forget how many he was using. He had to run the SNAPHU unwrapping on a computer with 256 GB of RAM to get it to avoid memory errors.

The ICU program was designed from the beginning to do the phase unwrapping in patches to avoid memory problems. It was written for processing the SRTM data that had very long strips that could never be loaded all into memory (even now, never mind in 2000 when SRTM flew).

**Reply by EJFielding** [reply]
The stand-alone version of SNAPHU from Stanford University has a tiling mode where it breaks the interferogram into tiles and unwraps each tile separately before combining them, but I don't know if it is possible to do that inside ISCE2. It is not available through any regular parameters you can set.

**Reply by ZachHoppinen** [reply]
Have there been comparisons between the two showing that SNAPHU is doing a better job? Ie should I go back through and increase my multilooking to use SNAPHU? Is there an advantage to that over just using icu?

I am using 8 looks in azimuth 2 in range, just based on Forrest's notebook so I will increase to 3x12 and see if I can get the snaphu to run if neccessary/useful.

**Reply by EJFielding** [reply]
Umm, there are many ways ICU and SNAPHU differ. I don't have time to explain. If you got good results with ICU, you can roll with that.

**Reply by ZachHoppinen** [reply]
Sounds good. I will multi-look and compare the two and see if there is a difference. Thanks again for your time Eric!

**Reply by ZachHoppinen** [reply]
So after running through this process the unwrapped interferograms I generated have almost exclusively values above zero. Is there something about the interpretation of these ICU unwrapped phase values I am missing that would make the unwrapped phase pixels all above zero? Or is there some step that I missed that caused this?

**Reply by EJFielding** [reply]
InSAR phases are always relative. Different unwrapping programs choose different places to put the "zero" for the unwrapped phase. I think SNAPHU uses the upper left corner and ICU uses some kind of average of the top of the image. Either way, there is always an unknown bias constant for the InSAR phase. You need other information to determine what is a more accurate zero level and what constant to add or subtract.
